diag_manager/diag_util.F90

```diff
@@ -2170,12 +2170,19 @@ CHARACTER(len=128) FUNCTION get_time_string(filename, current_time)
     CHARACTER(len=20) :: hr !< string of current time (output)
     CHARACTER(len=20) :: mi !< string of current time (output)
     CHARACTER(len=20) :: sc !< string of current time (output)
-    CHARACTER(len=128) :: filetail
-
+    CHARACTER(len=MAX_NAME_LENGTH) :: filetail
+    character(len=2) :: first_fms_percent
+    
     format = '("-",i*.*)'
     CALL get_date(current_time, yr1, mo1, dy1, hr1, mi1, sc1)
     len = LEN_TRIM(filename)
-    first_percent = INDEX(filename, '%')
+    do i=1,9
+       write(first_fms_percent,"('%',i1)") i 
+       first_percent = INDEX(filename, first_fms_percent)
+       if (first_percent > 0) then
+          exit
+       endif
+    enddo
```

Comment:
I don't think this will find the index of first percent of the string `xyz%8f%2g%1300%400` correctly; could you do something like

```fortran
first_percent = INDEX(filename, '%1')
do i=2,9
  write(first_fms_percent,"('%',i1)") i 
  first_percent_loc = INDEX(filename, first_fms_percent)
  if (first_percent_loc > 0) then
    if (first_percent == 0) then
      first_percent = first_percent_loc
    else
      first_percent = min(first_percent, first_percent_loc)
  endif
enddo
```

Also, are you sure you don't need to capture `%0`?

Reply:
Oh, you pushed changes while I was typing that :) I think this is what you want in `find_first_fms_percent()`

Reply:
I don't understand your loop - the one I wrote appears to work correctly now.

Reply:
I see now, you are correct - I will update the PR.   Thanks